adam-core/src/main/scala/org/bdgenomics/adam/rdd/sequence/SliceRDD.scala

```diff
@@ -0,0 +1,232 @@
+   *
+   * @param filePath Path to save files to.
+   * @param asSingleFile If true, saves output as a single file.
+   * @param lineWidth Hard wrap FASTA formatted slice at line width, default 60.
+   */
+  def saveAsFasta(filePath: String, asSingleFile: Boolean = false, lineWidth: Int = 60) {
+
+    def toFasta(slice: Slice): String = {
+      val sb = new StringBuilder()
+      sb.append(">")
+      sb.append(slice.getName)
+      Option(slice.getDescription).foreach(n => sb.append(" ").append(n))
+      sb.append(s" slice.getStart-slice.getEnd:slice.getStrand")
+      for (line <- Splitter.fixedLength(lineWidth).split(slice.getSequence)) {
+        sb.append("\n")
+        sb.append(line)
+      }
+      sb.toString
+    }
+
+    writeTextRdd(rdd.map(toFasta), filePath, asSingleFile)
+  }
+
+  /**
+   * Merge slices into sequences.
+   *
+   * @return Returns a SequenceRDD containing merged slices.
+   */
+  def merge(): SequenceRDD = {
+    def toSequence(slice: Slice): Sequence = {
+      Sequence.newBuilder()
+        .setName(slice.getName)
+        .setDescription(slice.getDescription)
+        .setAlphabet(slice.getAlphabet)
+        .setSequence(slice.getSequence)
+        .setLength(slice.getLength)
+        //.setAttributes(slice.getAttributes)
+        .build
+    }
+
+    def mergeSequences(first: Sequence, second: Sequence): Sequence = {
+      Sequence.newBuilder(first)
+        .setLength(first.getLength + second.getLength)
+        .setSequence(first.getSequence + second.getSequence)
+        .build
+    }
+
+    val merged: RDD[Sequence] = rdd
+      .sortBy(slice => (slice.getName, slice.getStart))
+      .map(slice => (slice.getName, toSequence(slice)))
+      .reduceByKey(mergeSequences)
+      .values
+
+    SequenceRDD(merged)
+  }
+
+  /**
+   * @param newRdd The RDD to replace the underlying RDD with.
+   * @return Returns a new SliceRDD with the underlying RDD replaced.
+   */
+  protected def replaceRdd(newRdd: RDD[Slice]): SliceRDD = {
+    copy(rdd = newRdd)
+  }
+
+  /**
+   * @param slice Slice to extract a region from.
+   * @return Returns a reference region that covers the entirety of the slice.
+   */
+  protected def getReferenceRegions(slice: Slice): Seq[ReferenceRegion] = {
+    Seq(ReferenceRegion(slice.getName, slice.getStart, slice.getEnd, slice.getStrand))
+  }
+
+  /**
+   * Writes an RDD to disk as text and optionally merges.
+   *
+   * @param rdd RDD to save.
+   * @param outputPath Output path to save text files to.
+   * @param asSingleFile If true, combines all partition shards.
+   */
+  private def writeTextRdd[T](rdd: RDD[T],
+                              outputPath: String,
+                              asSingleFile: Boolean) {
+    if (asSingleFile) {
+
+      // write rdd to disk
+      val tailPath = "%s_tail".format(outputPath)
+      rdd.saveAsTextFile(tailPath)
+
+      // get the filesystem impl
+      val fs = FileSystem.get(rdd.context.hadoopConfiguration)
```

Comment:
You'll want to use:

```
val path = new Path(outputPath)
val fs = path.getFileSystem(rdd.context.hadoopConfiguration)
```

Otherwise, this will fail if the path is not in the file system that is the `fs.defaultFs` from the Configuration. `Path` here is `org.apache.hadoop.fs.Path`.